File: RDD_Scala/WordCount.scala
import org.apache.spark.SparkContext
import org.apache.spark.SparkConf

val conf = new SparkConf().setAppName("WordCount").setMaster("local")
//val conf = new SparkConf().setAppName("WordCount").setMaster("spark://192.168.167.148:7077")
val sc = new SparkContext(conf)

val text = sc.textFile("C:/Spark/RDD_Scala/Input/WordCount.txt")
val counts = text.flatMap(line => line.split(" ")).map(word => (word, 1)).reduceByKey(_ + _)

// Sắp xếp kết quả theo key (từ) theo thứ tự từ điển
val sortedCounts = counts.sortByKey()

sortedCounts.saveAsTextFile("C:/Spark/RDD_Scala/Output/WordCount")
sortedCounts.foreach { case (word, count) =>
  println(s"$word: $count")}
sc.stop()

File: RDD_Python/WordCount.py
import findspark
findspark.init()

from pyspark import SparkConf
from pyspark import SparkContext

conf = SparkConf()
# conf.setMaster('spark://192.168.1.91:7077')
conf.setMaster('local')
conf.setAppName('WordCount')

# Khởi tạo SparkContext
sc = SparkContext(conf=conf)

# Đọc dữ liệu từ một tập tin văn bản
text = sc.textFile("C:/Spark/RDD_Python/Input/WordCount.txt")

# Chia các dòng thành các từ và tạo RDD mới
counts = text.flatMap(lambda line: line.split())\
    .filter(lambda word: len(word) > 0)\
    .map(lambda word: (word, 1))\
    .reduceByKey(lambda a, b: a + b)

# Sắp xếp kết quả theo key
sorted_counts = counts.sortByKey()

# Lưu kết quả ra file văn bản
sorted_counts.saveAsTextFile("C:/Spark/RDD_Python/Output/WordCount")
output = sorted_counts.collect()
for (word, count) in output:
    print("%s: %i" % (word, count))

# Dừng SparkContext
sc.stop()